### PR TITLE
Define KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION when requesting Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION=ON

### PR DESCRIPTION
**Pull request by masterleinad**
`KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION` was replaced by `KOKKOS_AGGRESSIVE_VECTORIZATION` in https://github.com/kokkos/kokkos/commit/c6844d1bb8ef36fa00301ee1e9cda54b9a0020c9 but it seems that we accidentally deleted the macro definition in https://github.com/kokkos/kokkos/commit/0591cf565a (and rather only defined the older name).

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
~~Considering~~
```
git grep -n KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION
Makefile.kokkos:629:  tmp := $(call kokkos_append_header,"$H""define KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION")
cmake/KokkosCore_config.h.in:56:#cmakedefine KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION
cmake/kokkos_enable_options.cmake:113:  SET(KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION ON)
```
~~I am leaning towards just removing.~~

**Comment by arnodu** [comment]
`#pragma ivdep` is correctly applied to my loop when using Intel 19, the Kokkos::OpenMP backend and with the `Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION` option in cmake now, thanks.

**Comment by dalg24** [comment]
> ANGE_AGGRESSIVE_VECTORIZATION

I take it back
```
git grep -n VECTORIZATION
BUILD.md:138:* Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION
Makefile.kokkos:73:KOKKOS_INTERNAL_OPT_RANGE_AGGRESSIVE_VECTORIZATION := $(call kokkos_has_string,$(KOKKOS_OPTIONS),aggressive_vectorization)
Makefile.kokkos:628:ifeq ($(KOKKOS_INTERNAL_OPT_RANGE_AGGRESSIVE_VECTORIZATION), 1)
Makefile.kokkos:629:  tmp := $(call kokkos_append_header,"$H""define KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION")
cmake/KokkosCore_config.h.in:56:#cmakedefine KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION
cmake/kokkos_enable_options.cmake:51:KOKKOS_ENABLE_OPTION(AGGRESSIVE_VECTORIZATION OFF "Whether to aggressively vectorize loops")
cmake/kokkos_enable_options.cmake:112:IF (KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION)
cmake/kokkos_enable_options.cmake:113:  SET(KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION ON)
core/cmake/KokkosCore_config.h.in:32:#cmakedefine KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
core/src/Cuda/Kokkos_Cuda_Vectorization.hpp:44:#ifndef KOKKOS_CUDA_VECTORIZATION_HPP
core/src/Cuda/Kokkos_Cuda_Vectorization.hpp:45:#define KOKKOS_CUDA_VECTORIZATION_HPP
core/src/Cuda/Kokkos_Cuda_Vectorization.hpp:232:#endif  // !defined( KOKKOS_CUDA_VECTORIZATION_HPP )
core/src/HIP/Kokkos_HIP_Vectorization.hpp:45:#ifndef KOKKOS_HIP_VECTORIZATION_HPP
core/src/HIP/Kokkos_HIP_Vectorization.hpp:46:#define KOKKOS_HIP_VECTORIZATION_HPP
core/src/Kokkos_Vectorization.hpp:47:#ifndef KOKKOS_VECTORIZATION_HPP
core/src/Kokkos_Vectorization.hpp:48:#define KOKKOS_VECTORIZATION_HPP
core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:80:#ifdef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:96:#ifdef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:175:#ifdef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
core/src/Threads/Kokkos_Threads_Parallel.hpp:83:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:98:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:190:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:395:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:410:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:580:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:867:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:882:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:945:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/Threads/Kokkos_Threads_Parallel.hpp:960:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \
core/src/impl/KokkosExp_Host_IterateTile.hpp:49:#if defined(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION) && \

File: cmake/KokkosCore_config.h.in [hunk 1]
```diff
@@ -53,7 +53,9 @@
 #cmakedefine KOKKOS_ENABLE_LARGE_MEM_TESTS
 #cmakedefine KOKKOS_ENABLE_DUALVIEW_MODIFY_CHECK
 #cmakedefine KOKKOS_ENABLE_COMPLEX_ALIGN
+// deprecated
 #cmakedefine KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION
```
Comment (dalg24) [inline cmake/KokkosCore_config.h.in:57]:
```suggestion
#cmakedefine KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION.  // deprecated
```

**Review by dalg24 — APPROVED** [review]

**Review by Rombur — APPROVED** [review]

**Pull request merged**